vignettes/IRangesOverview.Rmd

```diff
@@ -0,0 +1,527 @@
+subsequence. However, a sparser representation for consecutive subsequences
+would be a range, a pairing of a start position and a width, as used when
+extracting sequences with `window`.
+
+Two central classes are available in Bioconductor for representing ranges: the
+*IRanges* class defined in the `r Biocpkg("IRanges")` package for representing
+ranges defined on a single space, and the *GRanges* class defined in the 
+`r Biocpkg("GenomicRanges")` package for representing ranges defined on multiple
+spaces.
+
+In this vignette, we will focus on *IRanges* objects. We will rely on simple,
+illustrative example datasets, rather than large, real-world data, so that each
+data structure and algorithm can be explained in an intuitive, graphical manner.
+We expect that packages that apply `r Biocpkg("IRanges")` to a particular problem
+domain will provide vignettes with relevant, realistic examples.
+
+The `r Biocpkg("IRanges")` package is available at bioconductor.org and can be
+downloaded via `BiocManager::install`:
+
+```{r install, eval=FALSE, message=FALSE, warning=FALSE}
+if (!require("BiocManager"))
+    install.packages("BiocManager")
+BiocManager::install("IRanges")
+```
+
+```{r initialize, message=FALSE}
+library(IRanges)
+```
+
+#  *IRanges* objects
+
+To construct an object, we call the `IRanges` constructor. Ranges are normally
+specified by passing two out of the three parameters: start, end and width (see
+`help(IRanges)` for more information).
+
+```{r iranges-constructor}
+ir1 <- IRanges(start=1:10, width=10:1)
+ir1
+ir2 <- IRanges(start=1:10, end=11)
+ir3 <- IRanges(end=11, width=10:1)
+identical(ir1, ir2) && identical(ir1, ir3)
+ir <- IRanges(c(1, 8, 14, 15, 19, 34, 40),
+              width=c(12, 6, 6, 15, 6, 2, 7))
+ir
+```
+
+All of the above calls construct the same *IRanges* object, using different
+combinations of the `start`, `end` and `width` parameters.
+
+Accessing the starts, ends and widths is supported via the `start`, `end` and
+`width` getters:
+
+```{r iranges-start}
+start(ir)
+```
+
+```{r iranges-end}
+end(ir)
+```
+
+```{r iranges-width}
+width(ir)
+```
+
+Subsetting an *IRanges* object is supported, by numeric and logical indices:
+
+```{r iranges-subset-numeric}
+ir[1:4]
+```
+
+```{r iranges-subset-logical}
+ir[start(ir) <= 15]
+```
+
+In order to illustrate range operations, we'll create a function to plot ranges.
+
+```{r plotRanges}
+plotRanges <- function(x, xlim=x, main=deparse(substitute(x)),
+                       col="black", sep=0.5, ...) 
+{
+  height <- 1
+  if (is(xlim, "IntegerRanges"))
+    xlim <- c(min(start(xlim)), max(end(xlim)))
+  bins <- disjointBins(IRanges(start(x), end(x) + 1))
+  plot.new()
+  plot.window(xlim, c(0, max(bins)*(height + sep)))
+  ybottom <- bins * (sep + height) - height
+  rect(start(x)-0.5, ybottom, end(x)+0.5, ybottom + height, col=col, ...)
+  title(main)
+  axis(1)
+}
+```
+
+<a id="fig-ir-plotRanges"></a>
```

Comment:
I think these HTML anchors, like `<a id="fig-ir-plotRanges"></a>`, can be removed since there are no internal links to them within the document.

Reply:
I have noticed it too. The reason behind keeping these tags is that the author has mentioned labels in the .Rnw file, so I thought they may be used somewhere in the future. But I will remove them now.

Reply:
I just realized that these labels should be included in the chunk labels. I will take care of that and mark this conversation as unresolved.